### ISSUE TITLE
Ridiculous lag when certain entitites are on screen

### ISSUE THREAD

**Issue by Lonami** [Fargowilta/FargowiltasSouls#183]
The following items cause a tremendous amount of lag when used:

* Agitating lens. The scythes cause a lot of lag.
* Red spike balls. The items the goblins drop on death.
* Skeletron bones. Those he throws while spinning.

…and possibly many other. I play on a laptop, not a desktop PC, so I understand the performance may not be the best. But these items cause a lot of lag. I don't know why, or how they are implemented, but whatever they are doing is causing a lot of lag. I guess it's a performance issue in their implementation.

After searching in the forum for "laggy" or "performance" I have found this post https://forums.terraria.org/index.php?threads/fargos-mutant-mod-summons-and-souls.42291/page-167#post-1862400. My specs are nowhere close to that: 8GB RAM, Intel i5-5200U (4) @ 2.700GHz CPU, Intel HD Graphics 5500 integrated card, but the lag is still really bad.
I am not the only one to experience this lag. Other people I play with (also in a laptop) experience this issue too, and it occurs in both Linux and Windows.

In this gif you can see how, as soon as the bones despawn, speed is back to normal:
![a](https://user-images.githubusercontent.com/6297805/74053880-22488700-49dd-11ea-9d95-f685f36e395a.gif)

**Comment by Pbone3** [comment]
That is a known bug. For whatever reason, many masomode projectiles are bugged on Mac and Linux. Even weirder, changing the colour has occasionally fixed them in the past, but no "rule" has been found to fix them. To fix this issue, you can compile the mod yourself on Mac/Linux. I've never heard of the issue on windows, so I can't be sure weather or not compiling helps on windows. 

TL;DR: Known bug, but unknown how it's caused (I used to think it was a bug with Xna-To_Fna, a program used to compile xna code to fna), but this new report proves otherwise. Compiling it yourself may work, but that's only shown results on Mac/Linux, and it's unknown weather is solves the windows issue as well

**Comment by Lonami** [comment]
That's interesting, maybe a way to figure out how it breaks would be to add a lot of tracing log messages and compare the timestamps in each of the log lines when ran on Windows and compared to Linux.

**Comment by Fargowilta** [comment]
possibly fixed next update which is in 2 days

**Comment by Lonami** [comment]
Thanks, is there any particular commit fixing this or a merge?

**Comment by Fargowilta** [comment]
https://github.com/Fargowilta/FargowiltasSouls/commit/01ce595740a84bc175ff12d1e55700ac94b43c2a
